src/Controller/TideApiController.php

```diff
@@ -343,29 +343,29 @@ private function resolveAliasPath(Request $request, $path, $site, $cid, array &$
 
     // Dispatch a GET_ROUTE event so that other modules can modify it.
     if ($code != Response::HTTP_BAD_REQUEST) {
-      $event_entity = NULL;
-      if ($entity) {
-        $event_entity = clone $entity;
-      }
-      $event = new GetRouteEvent(clone $request, $json_response, $event_entity, $code);
+      $entity = $entity ?? NULL;
+      $event = new GetRouteEvent(clone $request, $json_response, (!empty($entity) ? clone $entity : NULL), $code);
```

Comment:
For my understanding, what we're trying to do here is to try and not create a hard dependency on `tide_site` from `tide_api` by doing a `dispatch` that `tide_site` can respond to if enabled?

This is more discussion than any change to make now, but I feel that we're not using the `$event_entity` parameter at all? These checks I think are redundant as looking at `onApiGetRouteAddSiteFilter` it will just populate the event anyway, so maybe it shouldn't be part of the constructor? Anyway, maybe flag for a wider discussion about how we should decouple our modules.

Reply:
`onApiGetRouteAddSiteFilter` has `$entity = $event->getEntity();` at line 69.

Reply:
we are passing a `clone` of the entity to the event subscribers, not the original `$entity`; hence the check is necessary.

Reply:
Yep, but my point is how is it different to the `entity` that it would get from the uuid anyway? All good, more just discussion points. I still think this could all do with a bit of a refactor but would need time in front of the whiteboard :). I think this PR is still fit for purpose.

Reply:
`$entity` is a fully loaded object. If it has the entity it doesn't need to lookup and load by UUID. Both `entity` and UUID are not always available in the event:
* If Tide API loads a cached response, there is no entity object, and Tide Site has to look up by UUID.
* Tide API may reject a path, hence no entity and UUID passed to Tide Site. But Tide Site accepts the path with the correct `site` param, in this case Tide Site looks up and loads the entity itself regardless the value passed from Tide API.